libs/components/packages/src/schematics/migrations/update-7/ag-grid-28.schematic.ts

```diff
@@ -0,0 +1,195 @@
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import {
+  NodeDependencyType,
+  addPackageJsonDependency,
+  getPackageJsonDependency,
+  removePackageJsonDependency,
+} from '@schematics/angular/utility/dependencies';
+
+import ignore from 'ignore';
+import { relative } from 'path';
+import semver from 'semver';
+
+export const UPDATE_TO_VERSION = '28.2.0';
+
+const UPDATE_LT_VERSION = '28.0.0';
+const ANY_MODULE = '@ag-grid-community/';
+const ENT_MODULE = '@ag-grid-enterprise/';
+const AG_GRID = 'ag-grid-community';
+const AG_GRID_ENT = 'ag-grid-enterprise';
+const withComponents = 'AgGridModule.withComponents';
+const warned: string[] = [];
+
+function warnOnce(message: string) {
+  if (!warned.includes(message)) {
+    warned.push(message);
+    console.warn(message);
```

Comment:
We should use Angular's context logger for these warnings.
```
(tree, context) => {
  context.logger.warn(...)
}
```

Reply:
Good catch. Switched.